sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -2050,4 +2051,93 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       )
     )
   }
+
+  // scalastyle:off nonascii
+  test("UTF8String validation") {
```

Comment:
Does this test the codegen part, or is it just testing that constant folding works?

Reply:
these unit tests cover both interpreted and codegen paths